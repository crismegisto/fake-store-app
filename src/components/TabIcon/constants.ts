import Routes from 'constants/routes';

interface Icons {
  [name: string]: string;
}

const ICONS: Icons = {
  [Routes.HomeNavigator]: 'home',
  [Routes.Support]: 'support-agent'
};

export default ICONS;
