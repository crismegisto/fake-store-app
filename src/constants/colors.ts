export const primary = '#606c38';
export const secondary = '#283618';
export const tertiary = '#fefae0';
export const lightGray = '#C9C9C9';
