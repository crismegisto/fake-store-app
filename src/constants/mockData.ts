export const mockProducts = [
  {
    category: "men's clothing",
    id: 1,
    title: '1Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops',
    price: 109.95,
    rating: {
      rate: 3.9,
      count: 120
    },
    image: 'https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg',
    description:
      'Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday'
  },
  {
    category: "men's clothing",
    id: 2,
    title: 'Mens Casual Premium Slim Fit T-Shirts',
    price: 22.3,
    rating: {
      rate: 3.9,
      count: 120
    },
    image: 'https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg',
    description:
      'Slim-fitting style, contrast raglan long sleeve, three-button henley placket, light weight & soft fabric for breathable and comfortable wearing. And Solid stitched shirts with round neck made for durability and a great fit for casual fashion wear and diehard baseball fans. The Henley style round neckline includes a three-button placket.'
  }
];
