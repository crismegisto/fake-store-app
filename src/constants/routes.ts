enum Routes {
  HomeNavigator = 'HomeNavigator',
  Home = 'Home',
  Detail = 'Detail',
  Support = 'Support'
}

export default Routes;
