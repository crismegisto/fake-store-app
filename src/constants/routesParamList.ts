import { Product } from 'interfaces/product';

import Routes from './routes';
export type RoutesParamList = {
  [Routes.HomeNavigator]: undefined;
  [Routes.Home]: undefined;
  [Routes.Detail]: Product;
  [Routes.Support]: undefined;
};
