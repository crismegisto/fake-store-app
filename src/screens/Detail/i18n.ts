import i18n from 'config/i18n';

i18n.addResources('en', 'DETAIL', {
  REVIEWS: '{{count}} Reviews'
});
