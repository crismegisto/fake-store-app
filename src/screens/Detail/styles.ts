import { StyleSheet } from 'react-native';
import { tertiary } from 'src/constants/colors';

const IMAGE_SIZE = 150;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: tertiary
  },
  title: {
    margin: 20,
    fontSize: 20,
    fontWeight: 'bold'
  },
  subContainer: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  image: {
    width: IMAGE_SIZE,
    height: IMAGE_SIZE,
    alignSelf: 'center'
  },
  review: {
    alignItems: 'center'
  },
  ratingContainer: {
    height: 10,
    marginBottom: 22
  },
  price: {
    margin: 14,
    color: 'green',
    fontSize: 16,
    fontWeight: 'bold'
  },
  description: {
    fontSize: 16,
    margin: 20
  }
});

export default styles;
