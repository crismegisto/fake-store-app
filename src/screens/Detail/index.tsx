import { RouteProp, useRoute } from '@react-navigation/native';
import React from 'react';
import { View, Text, ScrollView, Image } from 'react-native';
import { Product } from 'interfaces/product';
import notFound from 'assets/not-found.png';
import { AirbnbRating } from '@rneui/base';
import { useTranslation } from 'react-i18next';

import { RoutesParamList } from 'constants/routesParamList';

import './i18n';
import styles from './styles';

function Detail() {
  const { params } = useRoute<RouteProp<RoutesParamList>>();
  const { title, price, rating, image, description } = params as Product;
  const { t } = useTranslation('DETAIL');

  return (
    <View style={styles.container}>
      <ScrollView>
        <Text style={styles.title}>{title}</Text>

        <View style={styles.subContainer}>
          <Image source={image ? { uri: image } : notFound} resizeMode="contain" style={styles.image} />
          <View style={styles.review}>
            <AirbnbRating
              isDisabled
              defaultRating={rating.rate}
              size={16}
              reviewSize={10}
              ratingContainerStyle={styles.ratingContainer}
            />
            <Text testID="review">{t('REVIEWS', { count: rating.count })}</Text>
            <View>
              <Text style={styles.price}>${price}</Text>
            </View>
          </View>
        </View>

        <Text style={styles.description}>{description}</Text>
      </ScrollView>
    </View>
  );
}

export default Detail;
