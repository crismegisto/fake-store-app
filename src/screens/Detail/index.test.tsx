import React from 'react';
import { screen, render } from '@testing-library/react-native';

import { mockProducts } from 'constants/mockData';

import Detail from './index';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn()
    }),
    useRoute: () => ({ params: mockProducts[0] })
  };
});

jest.mock('@rneui/base', () => ({
  AirbnbRating: jest.fn()
}));

beforeEach(() => {
  render(<Detail />);
});

describe('Detail screen Tests', () => {
  test('match snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
