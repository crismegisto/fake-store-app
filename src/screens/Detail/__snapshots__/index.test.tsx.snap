// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail screen Tests match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#fefae0",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          {
            "fontSize": 20,
            "fontWeight": "bold",
            "margin": 20,
          }
        }
      >
        1Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            {
              "uri": "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
            }
          }
          style={
            {
              "alignSelf": "center",
              "height": 150,
              "width": 150,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
            }
          }
        >
          <Text
            testID="review"
          >
            120 Reviews
          </Text>
          <View>
            <Text
              style={
                {
                  "color": "green",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "margin": 14,
                }
              }
            >
              $
              109.95
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          {
            "fontSize": 16,
            "margin": 20,
          }
        }
      >
        Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
